Propose "why" explanation on discussions and decision making

potiuk opened the pull request:
This change is a proposal on how we could describe the next "why" that I consider as very important - discussions and decision making.

I tried to map all the red-flags we had for it into a coherent description of the "why" and linked to the two important places in the ASF "rules" that best describe some of the aspects of it.

I think the part of the variations might be contentious, and I hope that in the advisors "leaders" group we can discuss it and get to a consensus that

a) there are good reasons why the mailing lists were chosen in the
   long past

b) the reasons for that are still valid today and mailing list shoud
   remoain the most important place for discussions and decision making

while c) with the current set of technologies in many cases it is not the only way, but as long as PMC actively makes sure that the important discussions and decisions (and generally only those in more active projects) are brought to the list and where list is active and used for all the voting, super important decisions, this is all cool

Looking forward to discussion here.

justinmclean commented on wg-advisors/why/why_discussions_and_decisions.md:
```diff
@@ -1,13 +1,77 @@
+only be used for discussions that are absolutely not suitable for public consumption, such as discussions
+about security vulnerabilities, or discussions about individuals that are not suitable for public consumption
+and should be kept private. The private@ list should not be used for discussions about the project's direction,
+releases, or other important decisions.
```
One possible exception is discussing/voting on a release with a security issue.

potiuk replied:
Yep. Added.

justinmclean commented on wg-advisors/why/why_discussions_and_decisions.md:
```diff
@@ -1,13 +1,77 @@
+for all community members, allowing for asynchronous communication and ensuring that all community members
+have the opportunity to participate. All the important decisions and discussions happens there - including
+the decision to invite new committers and PMC members, the decisions to release software, and the decisions
+to change the project's direction.
```
Other good reasons for using mailing lists are that they are accessible to everyone in any country, can be archived, are easily searchable, and are asynchronous, so anyone can take part no matter their time zone/day job/other responsibilities. I can see They also have far more information density than online chat or some other forms of communication.

justinmclean replied:
Also filtered or grouped by rules.

potiuk commented:
Added some updates based on @justinmclean's comment. I really would love to hear from others, mainly about "using other tools as well". I think there are a number of "let's only use devlist" or "bring all the communication to the devlist automaticallly" - but as stated in the current version - there is a balance between "low signal-to-noise" on one end and "decisions made outside of the list" - which is I think the `gist` of the whole `why`. Too much in one direction is bad, too much in the other is also bad for other reasons. 

I wonder if the way described now is good in describing it.

rbowen commented:
There are two primary reasons we use mailing lists: 1) Accessibility and 2) Archive. That is, 1) ensuring that everyone in the world, regardless of language, timezone, or quality of network, can participate; and 2) ensuring that we have a permanent record of the discussion for later reference.

We didn't invent mailing lists, or even have any discussion about how it was to be done. That's just how things were done in 1994, and had been done that way for a decade leading up to that. We were standing on the shoulders of giants, and recognizing the already-acknowledged benefits. 

I don't see point two mentioned, but what you have written is fine. +1 to merging what you have. Possibly a mention of the archival nature of mailing lists would be warranted.

potiuk commented:
> I don't see point two mentioned

There was a paragraph on that already:

> The mailing list provide a permanent record of the discussions and decisions and the Apache Software Foundation
Infrastructure team makes sure that the archives are preserved and available for the future, including the
ability to search and browse the archives, as well as permanently link to the discussions and decisions so
that they can be always referred to in the future.

But I added a one sentence summary at the top and updated the paragraph to stress importance of it. The description indeed becomes a bit long to read it in it's entirety and get the quick grasp of what's in, so having a summarizing one line at the beginning is a good idea.
